ARCs/arc-0004.md

```diff
@@ -143,12 +143,30 @@ interface Method {
     /** Optional, user-friendly description for the argument */
     desc?: string
   }>,
+  /** A list of allowed OnCompletion values for the method. Must not be empty */
+  allowedActions: OnCompletionAction[],
   /** Information about the method's return value */
   returns: { type: string, desc?: string }
 }
 ```
 
-For example:
+Where `OnCompletionAction` is defined as:
+
+```typescript
+type OnCompletionAction = "noop" | "optin" | "closeout" | "clearstate" | "update" | "delete";
+```
+
+Each string value represents an application call transaction OnCompletion
+(`apan`) value. The mapping from these string values to the numeric values
+for transaction encoding is:
```

Comment:
Should we link directly to the go source file that defines these?

Reply:
yes. I would

Reply:
Good suggestion, I'll add a link tomorrow